File: Tp-01/aleatoria.c
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <time.h>

int final(char str[])
{
    int tam = strlen(str);
    if (tam == 3)
    {
        if (str[0] == 'F')
        {
            if (str[1] == 'I')
            {
                if (str[2] == 'M')
                {
                    return 1;
                }
            }
        }
    }
    return 0;
}

char *bagunca(char *chars, char pri, char sec, int tam, int i)
{
    // verificar
    if (i == tam)
    {
        return chars;
    }
    // bagunçar
    if (chars[i] == pri)
    {
        chars[i] = sec;
    }
    // final
    return bagunca(chars, pri, sec, tam, (i + 1));
}

int main()
{
    // definir dados
    char *input = (char *)malloc(1001 * sizeof(char));
    char *resp = (char *)malloc(1001 * sizeof(char));
    char pri;
    char sec;
    int fim = 0;
    // gerador
    srand(time(NULL));
    // ler primeiro input
    scanf("%[^\n]%*c", input);
    int tam = strlen(input) - 1;
    fim = final(input);
    // rodar enquanto nao for 'FIM'
    while (fim == 0)
    {
        pri = ((char)('a' + rand() % 26));
        sec = ((char)('a' + rand() % 26));
        resp = bagunca(input, pri, sec, tam, 0);
        printf("%s\n", resp);
        scanf("%[^\n]%*c", input);
        tam = strlen(input) - 1;
        fim = final(input);
    }
    return 0;
}

File: Tp-01/Aleatoria.java
import java.util.*;

public class Aleatoria {
    static String bagunca(String input, char pri, char sec) {
        // definir dados
        char[] chars = input.toCharArray();

        // bagunçar
        for (int i = 0; i < input.length(); i++) {
            if (input.charAt(i) == pri) {
                chars[i] = sec;
            }
        }
        // final
        return new String(chars);
    }

    public static void main(String[] args) {
        String input = "";
        String resp = "";
        char pri;
        char sec;
        // gerador
        Random gerador = new Random();
        gerador.setSeed(4);
        // Scanner scanner = new Scanner(System.in);
        // input = scanner.nextLine();
        input = MyIO.readLine();
        while (!input.equals("FIM")) {
            pri = ((char) ('a' + (Math.abs(gerador.nextInt()) % 26)));
            sec = ((char) ('a' + (Math.abs(gerador.nextInt()) % 26)));
            resp = bagunca(input, pri, sec);
            MyIO.println(resp);
            input = MyIO.readLine();
        }
    }
}
